acceptance/mockAgent/hostConfig.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/control-center/serviced/utils"
+	"github.com/control-center/serviced/validation"
+)
+
+type HostConfig struct {
+		Name       string   `json:"name"`
+		RPCPort    int      `json:"rpcPort"`
+		PoolID     string   `json:"pool,omitempty"`
+		Memory     string   `json:"memory,omitempty"`
+		HostID     uint16   `json:"hostID,omitempty"`
+		OutboundIP string   `json:"outboundIP,omitempty"`
+		StaticIPs  []string `json:"staticIPs,omitempty"`
+		Listen     string   `json:"-"`
+}
+
+func (hostConfig *HostConfig) setDefaults() error {
+	if hostConfig.Listen == "" {
+		hostConfig.Listen = fmt.Sprintf(":%d", hostConfig.RPCPort)
+	}
+
+	if hostConfig.HostID == 0 {
```

Comment:
Since we now have HostID in the configuration data, I think we can treat that has a required attribute and remove lines41-43 and `randomHostId()` to avoid any potential confusion about where it comes from